Templates/DefaultProject/Template/ExportScripts/export_standalone_monolithic.py

```diff
@@ -0,0 +1,321 @@
+import shutil
+import sys
+import time
+import psutil
+
+from o3de.export_project import process_command
+from o3de import manifest
+from typing import List
+
+
+# This script is meant to be run via o3de cli export-project command, not from Python directly!
+# This export script operates in a project-centric manner, meaning all build artifacts are centered around the project folder. 
+# In order to use this script, the command should look like the following (on windows use o3de.bat, on linux use o3de.sh):
+
+# cd path\to\o3de\engine
+# .\scripts\o3de.bat  export-project --export-scripts path\to\o3de\ExportScripts\export_standalone_monolithic_project_centric.py \
+# --project-path path\to\project --log-level INFO --output-path path\to\output --build-non-mono-tools --config release --archive-output zip \
+# --seedlist path\to\seedlist1.seed --seedlist path\to\seedlist2.seed <etc..> 
+
+# The following parameters are optional, but may be helpful:
+#   seedlist: an O3DE seedlist for describing what assets of the project should be bundled
+#   gem-folder-name-to-copy: A Gem Folder from <ProjectRoot>/Gems that should be copied to the output location of the exported project
+#                               Files found in these folders will go into <output-path>/Gems/GemFolderName
+#                               Variants for game/server launcher only are game-gem-folder-name-to-copy/server-gem-folder-name-to-copy resp.
+#   project-file-pattern-to-copy: Any regex pattern starting from <ProjectRoot> that indicates what files should be copied to output location.
+#                                   Files found in these patterns will go into <output-path>
+#                                   Variants for game/server launcher only are game-file-pattern-to-copy/server-file-pattern-to-copy resp.
+
+# For more information on the available parameters for this script, check the parse_arguments function defined at the bottom of the file
+
+def  build_non_monolithic_code_modules( project_path: pathlib.Path,
+                                        non_mono_build_path: pathlib.Path):
+    #build the pre_requisite tools if needed
+    os.makedirs(non_mono_build_path, exist_ok=True)
+
+    process_command(["cmake", "-B", non_mono_build_path, "-S", project_path])
+
+    process_command(["cmake", "--build", non_mono_build_path, "--target", "AssetProcessorBatch", "AssetBundlerBatch", "--config", "profile", "--", "/m"])
+
+def build_monolithic_code_modules( project_path: pathlib.Path,
+                        project_name: str,
+                        build_config: str,
+                        mono_build_path: pathlib.Path,
+                        should_build_game_launcher: bool,
+                        should_build_server_launcher: bool,
+                        should_build_unified_launcher: bool):
+    process_command(["cmake", "-B", mono_build_path, "-S", project_path, "-DLY_MONOLITHIC_GAME=1", "-DALLOW_SETTINGS_REGISTRY_DEVELOPMENT_OVERRIDES=0"])
```

Comment:
The [ALLOW_SETTINGS_REGISTRY_DEVELOPMENT_OVERRIDES](https://github.com/o3de/o3de/blob/development/Code/Framework/AzCore/CMakeLists.txt#L23-L24) should be configurable by the user.

They weren't given a choice as to whether they want to settings registry files to load from external sources when they ran the script.
I recommended adding an argparse option which defaults to `False` for the Cache variable